### DISCUSSION TITLE
How to define a custom loss for an inversion problem when the observed data is not directly the solution of the PDE?

### DISCUSSION THREAD

**Discussion by zjuliu** [lululxvi/deepxde#1845] [Q&A]
Thanks a lot for sharing such a powerful method. 
I'm using DeepXDE to invert the parameters in the following PDE equation:
![pde](https://github.com/user-attachments/assets/c2cd42da-4d8b-4f03-abf7-f1e17213984f)
where **_u_** = **_u_**(_a_, _t_), k = k(**_u_**),  σ' = σ'(**_u_**); other variables are constant. The constant parameters in k(**_u_**) and σ'(**_u_**) need to be inverted.

My observed data is not **_u_** but the a function of **_u_**, which is an array with shape of (n, 2) like
[t, f(**_u_**(a,t))], t = t0, t1, t2, ....
In my custom loss function, I want to extract the predicted **𝑢**(𝑎,𝑡) around the observed times, apply the function 𝑓(**𝑢**(𝑎,𝑡)), and then compare the result with the observed data. **But some errors always occurs. The time index always exceeds the dimension of y_pred in my custom loss function.**  
How can I solve this problem, and is there a more efficient way to achieve the same goal? Thanks a lot!
My code is as follows:

![code1](https://github.com/user-attachments/assets/08788ef2-18ab-46ce-aea6-18be2b11c263)
![code2](https://github.com/user-attachments/assets/b7120481-283a-4c0c-8480-9548912cc07f)
![code3](https://github.com/user-attachments/assets/46601455-7ea0-487e-a2d4-2c8afbb12f18)

**Comment by lululxvi** [comment]
Let v = f(u(a,t)) be another PDE equation, although this is not really PDE. Then data is for v.
